Remove legacy v1 kubeflow pipelines provider (#748)
sky-uk/kfp-operator: docs-gen/content/docs/getting-started/introduction.md

The operator currently supports

> - TFX Pipelines with Python 3.7 and 3.9 - pipelines created using the KFP DSL are not supported yet

Reviewer: this is now supported, I know not related to this PR but can you update it? 
```suggestion
- TFX Pipelines with Python 3.7 and 3.9
- [Kubeflow Pipelines SDK](https://kubeflow-pipelines.readthedocs.io/)
```

Author: https://github.com/sky-uk/kfp-operator/pull/748/commits/8b63d4866e99077bf629081a88da812ecfe33141
